File: python/백준/정렬/18870 좌표 압축.py
n = int(input())  # 배열의 크기 입력
array = list(map(int, input().split()))  # 배열 입력

# 좌표 압축 수행
array_s = sorted(set(array))  # 중복 제거 후 정렬
dic = {array_s[i]: i for i in range(len(array_s))}  # 값에 순위를 매핑

# 압축된 값 출력
for a in array:
    print(dic[a], end=" ")

File: cpp/백준/정렬/18870 좌표 압축.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <unordered_map>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    vector<int> array(n);
    for (int i = 0; i < n; ++i) cin >> array[i];

    vector<int> array_s(array);
    sort(array_s.begin(), array_s.end());
    array_s.erase(unique(array_s.begin(), array_s.end()), array_s.end());

    unordered_map<int, int> dic;
    for (int i = 0; i < (int)array_s.size(); ++i) {
        dic[array_s[i]] = i;
    }

    for (int a : array) {
        cout << dic[a] << ' ';
    }
    cout << "\n";
    return 0;
}
